### PR TITLE
Ci test book

### LINKED ISSUE #1502: [MAINTENANCE] Update CI to test book
## Description

Concourse CI isn't testing the book ([concourse task](https://github.com/amethyst/amethyst/blob/master/.concourse/tasks/linux-test.yml)). This [used to be done](https://github.com/amethyst/amethyst/blob/master/.travis.yml#L70) in Travis CI.

## Reason

This keeps the book up to date as the code evolves.

## Impact

Would increase build times slightly.

### PR DESCRIPTION
**Pull request by azriel91**
## Description

Fixes #1502 (test book in CI), supersedes #1587.

## PR Checklist

By placing an x in the boxes I certify that I have:

- **n/a** Ran `cargo test --all` locally if this modified any rs files.
- **n/a** Ran `cargo +stable fmt --all` locally if this modified any rs files.
- **n/a** Updated the content of the book if this PR would make the book outdated.
- **n/a** Added a changelog entry if this will impact users, or modified more than 5 lines of Rust that wasn't a doc comment.
- **n/a** Added unit tests for new APIs if any were added in this PR.
- [x] Acknowledged that by making this pull request I release this code under an MIT/Apache 2.0 dual licensing scheme.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -92,18 +92,23 @@ pipeline {
                         }
                     }
                     steps {
+                        echo 'Cleaning up old build artifacts'
+                        sh 'rm -rf ./target/debug/deps/libamethyst* || true'
```
Comment (torkleyy) [inline Jenkinsfile]: This also doesn't work reliably, because as soon as we have a new patch version in any of our dependencies, `mdbook` will fail again.

Comment (azriel91) [reply Jenkinsfile]: yeap, just realized; I'm crafting a bash script, which isn't great, but I think it's worth testing the book

**Comment by azriel91** [comment]
Crates with multiple lib artifacts in my local machine (`echo`ed):

```bash
$ ./scripts/remove_duplicate_libs.sh  | wc -l
323

# Without `uniq` in the script:
$ ./scripts/remove_duplicate_libs.sh  | wc -l
3143
```

**Comment by azriel91** [comment]
ah, still get `multiple matching crates`, okay I'm convinced.

**Pull request closed without merging**